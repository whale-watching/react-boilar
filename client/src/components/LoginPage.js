import React from 'react';
import { socketEmit } from '../helpers/socketEvents';

class LoginPage extends React.Component {
  constructor() {
    super();

    this.state = {
      error: null,
    };

    this.loginUser = (e) => {
      e.preventDefault();

      const userName = e.target.elements.userName.value.trim();

      if (!userName) {
        return this.setState({ error: 'You must enter username' });
      }

      socketEmit.joinUser(userName, (err) => {
        this.setState({ error: err });
      });

      e.target.elements.userName.value = '';
    };
  }

  render() {
    return (
      <div className="login-page">
        <div className="login-modal">
          <form onSubmit={this.loginUser}>
            <h3>Join Chat</h3>
            <p className="error">{this.state.error}</p>
            <p>Username</p>
            <input type="text" name="userName" maxLength="20" autoFocus autoComplete="off" />
            <button type="submit" className="button-text">Join</button>
          </form>
        </div>
      </div>
    );
  }
}

export default LoginPage;
