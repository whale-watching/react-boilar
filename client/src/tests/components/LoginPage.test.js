import React from 'react';
import { shallow } from 'enzyme';
import LoginPage from '../../components/LoginPage';
import loginSubmit from '../data/loginSubmit';

describe('<LoginPage />', () => {
  it('renders LoginPage correctly', () => {
    const wrapper = shallow(<LoginPage />);

    expect(wrapper).toMatchSnapshot();
  });

  it('renders error if no username is provided', () => {
    const wrapper = shallow(<LoginPage />);

    wrapper.find('form').simulate('submit', loginSubmit);

    expect(wrapper.state('error')).toBe('You must enter username');
  });
});