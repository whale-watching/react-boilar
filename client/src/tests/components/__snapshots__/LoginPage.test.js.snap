// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginPage /> renders LoginPage correctly 1`] = `
<div
  className="login-page"
>
  <div
    className="login-modal"
  >
    <form
      onSubmit={[Function]}
    >
      <h3>
        Join Chat
      </h3>
      <p
        className="error"
      />
      <p>
        Username
      </p>
      <input
        autoComplete="off"
        autoFocus={true}
        maxLength="20"
        name="userName"
        type="text"
      />
      <button
        className="button-text"
        type="submit"
      >
        Join
      </button>
    </form>
  </div>
</div>
`;
